src/lib.rs

```diff
@@ -141,13 +146,18 @@ macro_rules! impl_file_size_u {
     			};
 
     			if opts.long_suffix && size.trunc() == 1.0 { scale = &scale[0 .. scale.len()-1];}
-			
+
     			let places = match size.fract() {
     				0.0 => opts.decimal_zeroes,
     				_ => opts.decimal_places
     			};
-			
-    			Ok(format!("{:.*} {}", places, size, scale))
+
+                let mut space = String::new();
+                if opts.space {
+                    space.push(' ');
```

Comment:
To be more idiomatic, this could be written as a match block or an if block. (In case you didn't know, if-blocks in rust are expressions and thus can return a value.)
E.g.

```
let space = if opts.space {
    " "
} else {
    ""
};
```

OR

```
let space = match opts.space {
    true => " ",
    false => ""
};
```
In both cases, we avoid the need for explicitly allocating a String, which is why I think it looks nicer!